homeassistant/components/tuya/light.py

```diff
@@ -432,9 +443,10 @@ def __init__(
         if self._color_data_dpcode:
             self._attr_supported_color_modes.add(COLOR_MODE_HS)
             # Fetch color data type information
-            if function_data := json.loads(
+            function_data = json.loads(
                 self.device.function[self._color_data_dpcode].values
-            ):
+            )
+            if hasattr(function_data, "h"):
```

Comment:
Why do we need to check this?

Reply:
This is the `function` block of the data on my device:

```
    function={
        "switch": TuyaDeviceFunction(code="switch", type="Boolean", values="{}"),
        "switch_spray": TuyaDeviceFunction(
            code="switch_spray", type="Boolean", values="{}"
        ),
        "mode": TuyaDeviceFunction(
            code="mode", type="Enum", values='{"range":["large","middle","small"]}'
        ),
        "countdown": TuyaDeviceFunction(
            code="countdown", type="Enum", values='{"range":["cancel","1","3","6"]}'
        ),
        "switch_led": TuyaDeviceFunction(
            code="switch_led", type="Boolean", values="{}"
        ),
        "bright_value": TuyaDeviceFunction(
            code="bright_value",
            type="Integer",
            values='{"unit":"","min":10,"max":1000,"scale":0,"step":1}',
        ),
        "work_mode": TuyaDeviceFunction(
            code="work_mode",
            type="Enum",
            values='{"range":["white","colour","colourful1"]}',
        ),
        "colour_data_hsv": TuyaDeviceFunction(
            code="colour_data_hsv",
            type="Json",
            values='{"min":0,"unit":"","scale":0,"max":255,"step":1}',
        ),
    },
```

The `colour_data_hsv` block doesn't have seperate values for `h`, `s` & `v`, so I was getting a `KeyError` without this.

I tried with the following but since the hue value is 0-360, not 0-255, the colour wheel broke

```
h_type=IntegerTypeData(**getattr(function_data, "h", function_data))
```

If you've got a better idea I'd be happy to hear it, I just went with something I felt would have the least chance of breaking other devices.

Reply:
> The colour_data_hsv block doesn't have seperate values for h, s & v

It should have that. Did you report this issue with Tuya? As this sounds like an issue/bug upstream.